Fix compiler warnings[RDY]

aravindprasads opened the pull request:
Fixes the following warning during compilation:
---
1) config.c: In function 'rc_new':
config.c:1108:6: warning: unused variable 'ret' [-Wunused-variable]
  int ret;
      ^
2) common.c:87:2: warning: ignoring return value of ‘write’, declared with attribute warn_unused_result [-Wunused-result]
  (void)write(out, prompt, strlen(prompt));
  ^
3) common.c:111:5: warning: ignoring return value of ‘write’, declared with attribute warn_unused_result [-Wunused-result]
     (void)write(out, &c, 1);
     ^
4) common.c:118:28: warning: ignoring return value of ‘write’, declared with attribute warn_unused_result [-Wunused-result]
  if (!do_echo || !is_term) (void)write(out, "\r\n", 2);

5) tls-restart.c: In function ‘process’:
tls-restart.c:161:6: warning: ignoring return value of ‘dup’, declared with attribute warn_unused_result [-Wunused-result]
   dup(fd);
      ^

aravindprasads committed:
Fix compiler warnings

nmav commented on src/common.c:
```diff
@@ -84,7 +84,9 @@ char *rc_getstr (rc_handle *rh, char const *prompt, int do_echo)
 	}
 
-	(void)write(out, prompt, strlen(prompt));
+	if(write(out, prompt, strlen(prompt)) == -1) {
```
nit: identation (space after if)

aravindprasads replied:
Addressed

nmav commented on lib/config.c:
```diff
@@ -1107,6 +1107,7 @@ rc_handle *rc_new(void)
 	rc_handle *rh;
 	int ret;
 
+	(void)ret;
```
What about this re-arrangement?
```
@@ -1105,11 +1105,10 @@ static int _initialized = 0;
 rc_handle *rc_new(void)
 {
        rc_handle *rh;
-       int ret;
 
        if (_initialized == 0) {
-               srandom((unsigned int)(time(NULL)+getpid()));
 #if defined(HAVE_GNUTLS) && GNUTLS_VERSION_NUMBER < 0x030300
+               int ret;
                ret = gnutls_global_init();
                if (ret < 0) {
                        rc_log(LOG_ERR,
@@ -1118,6 +1117,7 @@ rc_handle *rc_new(void)
                        return NULL;
                }
 #endif
+               srandom((unsigned int)(time(NULL)+getpid()));
        }
        _initialized++;
 
````

aravindprasads replied:
Yes, agreed and addressed

nmav commented on src/common.c:
```diff
@@ -107,15 +109,24 @@ char *rc_getstr (rc_handle *rh, char const *prompt, int do_echo)
 
-	if (!do_echo || !is_term) (void)write(out, "\r\n", 2);
+	if (!do_echo || !is_term) {
+		if(write(out, "\r\n", 2) == -1) {
```
nit: indentation.
However here the error is also return values < 2. Shouldn't we exit when these errors are detected? That's example applications so it shouldn't be a big deal.

aravindprasads replied:
Addressed. I think we need not exit and carry on with processing.

nmav reviewed (commented):
Thank you. Except for the comments it looks ok to me.

aravindprasads commented:
Hi @nmav ,

Thanks for the review. I have addressed the comments. Kindly review the same. 

Also, added a change to check return value for dup call.

nmav commented:
Thank you!
